### DISCUSSION TITLE
Allow plugin manager read addons folder inside subfolders.

### DISCUSSION THREAD

**Discussion by jinsediaoying** [godotengine/godot-proposals#8100] [Editor]
This is similar to unity's Editor and Resources folder behaviour.
By doing so, we can organize codes by module.

like this
![image](https://github.com/godotengine/godot-proposals/assets/25916527/60e08a06-05ae-4924-be49-f80d639ed62c)
 instead of currently this
![image](https://github.com/godotengine/godot-proposals/assets/25916527/3163566e-5811-41a3-b923-864671faf6f2)

**Comment by AnidemDex** [comment]
By default, Editor only look for plugin.cfg for each folder in `addons`, but you can activate/deactivate plugins in subfolders using the EditorInterface. Does that help? :thinking:

**Reply by jinsediaoying** [reply]
My problem is that I want to create some code packs that I can easily share between projects (maybe as a git submodule).
Each code pack will have its own editor plugin, so I prefer to put all of stuff into one single folder.

Ops, I just realized I can make res://addons/MySystem to be that only folder!

**Reply by jinsediaoying** [reply]
It works, but it also mean I have to put all my runtime codes inside addons folder too.
I'm not sure it is a good practice, since addons folder more feels like a editor only stuff.

**Reply by AnidemDex** [reply]
addons are usually editor + game tools stuff
